### ISSUE TITLE
Configuration warning

### ISSUE THREAD

**Issue by akzincsystems** [dougmoscrop/serverless-plugin-log-subscription#18]
sls deploy gives warning

Configuration warning at 'functions.indexertest': unrecognized property 'logSubscription'

As the docs say, it'll bypass at the moment and run, but future release of serverless may stop deployment for such things.

**Comment by dougmoscrop** [comment]
Yeah I don't know what the plan is around this validation, it will make a lot of plugin authors frustrated and plugin consumers too - because you'd have to re-define all your function names elsewhere (custom)

I think a logical solution would be to support `custom` on functions

**Comment by panumjp** [comment]
How about defining `defineFunctionProperties` in constructor? https://www.serverless.com/framework/docs/providers/aws/guide/plugins#extending-validation-schema

```
    serverless.configSchemaHandler.defineFunctionProperties(this.provider, {
      properties: {
        logSubscription: { type: 'boolean' }
      },
    });
```

Seems to work, but I don't know what is the best way to handle this in existing unit tests.

**Comment by medikoo** [comment]
> Yeah I don't know what the plan is around this validation, it will make a lot of plugin authors frustrated and plugin consumers too - because you'd have to re-define all your function names elsewhere (custom)

@dougmoscrop it's not how it works. As @panumjp above pointed, there's dedicated `defineFunctionProperties`, through which easily (any function) properties can be defined. See: https://www.serverless.com/framework/docs/providers/aws/guide/plugins#definefunctionproperties-helper

I believe addressing this deprecation is relatively easy here

**Comment by dougmoscrop** [comment]
Thanks, fixed in 2.0.0

**Comment by medikoo** [comment]
That's great news. Thank you @dougmoscrop !

**Comment by chippyash** [comment]
@dougmoscrop Belated thanks for fixing this.

**Comment by seanthingee** [comment]
@dougmoscrop 
> Thanks, fixed in 2.0.0

I'm having this issue in 2.1.2. I'm using serverless 3.16.0. Like others have noted it appears to deploy fine.
